command/agent/auth/auth.go

```diff
@@ -156,6 +157,15 @@ func (ah *AuthHandler) Run(ctx context.Context, am AuthMethod) error {
 		credCh = make(chan struct{})
 	}
 
+	if ah.client != nil {
+		headers := ah.client.Headers()
+		if headers == nil {
+			headers = make(map[string][]string)
```

Comment:
Can we use "headers = make(http.Header)" here?

Reply:
Thanks for the suggestion! I guess I must have missed this then copied it throughout. Adding a PR with the fix.